shell/src/main/java/alluxio/cli/validation/StorageSpaceValidationTask.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.util.FormatUtils;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Task for validating whether worker tiered storage has enough space.
+ */
+public final class StorageSpaceValidationTask implements ValidationTask {
+
+  /**
+   * Creates a new instance of {@link StorageSpaceValidationTask}
+   * for validating tiered storage space.
+   */
+  public StorageSpaceValidationTask() {
+  }
+
+  @Override
+  public boolean validate() {
+    int numLevel = Configuration.getInt(PropertyKey.WORKER_TIERED_STORE_LEVELS);
+    boolean success = true;
+
+    for (int level = 0; level < numLevel; level++) {
+      PropertyKey tierAliasConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_ALIAS.format(level);
+      String alias = Configuration.get(tierAliasConf);
+
+      PropertyKey tierDirPathConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_PATH.format(level);
+      String[] dirPaths = Configuration.get(tierDirPathConf).split(",");
+
+      PropertyKey tierDirCapacityConf =
+          PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA.format(level);
+      String rawDirQuota = Configuration.get(tierDirCapacityConf);
+      if (rawDirQuota.length() <= 0) {
+        System.err.format("Tier %d: Quota cannot be empty.%n", level);
+        return false;
+      }
+
+      String[] dirQuotas = rawDirQuota.split(",");
+
+      try {
+        Map<String, MountedStorage> storageMap = new HashMap<>();
+        File file = new File(dirPaths[0]);
+        if (dirPaths.length == 1 && alias.equals("MEM") && !file.exists()) {
+          // skip checking if RAM disk is not mounted
+          System.out.format("RAM disk is not mounted at %s, skip validation.%n", dirPaths[0]);
+          continue;
+        }
+
+        for (int i = 0; i < dirPaths.length; i++) {
+          int index = i >= dirQuotas.length ? dirQuotas.length - 1 : i;
+          long quota = FormatUtils.parseSpaceSize(dirQuotas[index]);
+          success &= addDirectoryInfo(dirPaths[i], quota, storageMap);
+        }
+
+        for (Map.Entry<String, MountedStorage> storageEntry : storageMap.entrySet()) {
```

Comment:
From what I understand Alluxio worker will monitor the use of space on each directory so it does not go over the quota. 

Regarding this check, we would ideally want to ensure that each directory specified in the tier configuration have enough space to store Alluxio data with total size up to the given quota. However if user specifies two directories that are backed by same mounted storage, the space that are available to be used by Alluxio server is shared. 

For example, if `/` is the mount root with 3TB free space and user defines two directories `/tmp/` and `/home/alluxio_storage` for tier 1 storage with 2TB quota each, the server can run into issues because it cannot store 2TB of files under each directory, given that there is only 3TB on `/`. In this case, the best we can do is to check that the disk mounted on `/` has at least 2TB + 2TB free space so that Alluxio worker can store up to the quota for each directory without problem.